src/full/Agda/Compiler/Backend.hs

```diff
@@ -178,35 +180,44 @@ parseBackendOptions backends argv opts0 =
       opts <- checkOpts opts
       return (forgetAll forgetOpts backends, opts)
 
-backendInteraction :: AbsolutePath -> [Backend] -> TCM () -> (AbsolutePath -> TCM (Maybe Interface)) -> TCM ()
+backendInteraction :: AbsolutePath -> [Backend] -> TCM () -> (AbsolutePath -> TCM CheckResult) -> TCM ()
 backendInteraction mainFile backends setup check = do
   setup
-  mi <- check mainFile
+  checkResult <- check mainFile
 
   -- reset warnings
   stTCWarnings `setTCLens` []
 
   noMain <- optCompileNoMain <$> pragmaOptions
   let isMain | noMain    = NotMain
              | otherwise = IsMain
-  case mi of
-    Nothing -> genericError $ "You can only compile modules without unsolved metavariables."
-    Just i  -> sequence_ [ compilerMain backend isMain i | Backend backend <- backends ]
+
+  unlessM (optAllowUnsolved <$> pragmaOptions) $ do
+    let ws = crWarnings checkResult
+        mode = crMode checkResult
+    -- Possible warnings, but only scope checking: ok.
+    -- (Compatibility with scope checking done during options validation).
+    unless (mode == ModuleScopeChecked || null ws) $
+      genericError $ "You can only compile modules without unsolved metavariables."
```

Comment:
Do we need this check?
How do I trigger this error?

Reply:
The error message seems to have been introduced in 6698cc19cb87808b92311af3bf0698d8fe9a320a ("turn an \_\_IMPOSSIBLE__ into error when compiling allowed unsolved metas (inspired by #2474)"), but the logic might no longer be the same.

Reply:
I deleted the check.